shared_utils/s3_utils.py

```
@@ -96,6 +96,40 @@ def delete_file(key: str) -> bool:
         return False
 
 
+def list_folder(folder_prefix: str) -> list[str]:
+    paginator = s3_client.get_paginator("list_objects_v2")
+    pages = paginator.paginate(Bucket=S3_BUCKET, Prefix=folder_prefix)
+    keys = [obj["Key"] for page in pages for obj in page.get("Contents", [])]
+    return keys
+
+
+def delete_folder(folder_prefix: str) -> bool:
+    """
+    Deletes a folder from S3 using the given key.
+    Returns True if the folder was deleted, False if an error occured.
+    """
+    DELETE_OBJECT_LIMIT = 1000
+    try:
+        keys = list_folder(folder_prefix)
+        chunked_keys = [
+            keys[i : i + DELETE_OBJECT_LIMIT]
+            for i in range(0, len(keys), DELETE_OBJECT_LIMIT)
+        ]
+        for chunk in chunked_keys:
+            s3_client.delete_objects(
+                Bucket=S3_BUCKET,
+                Delete={"Objects": [{"Key": key} for key in chunk], "Quiet": True},
+            )
+        return True
+    except (BotoCoreError, ClientError) as e:
+        logger.exception(f"Failed to delete file from S3: {e}")
+        return False
```

Not sure if this function needs a catch all exception

save_job() has a catch all exception (but no `except (BotoCoreError, ClientError) as e:`)

---

Will leave for for enhancement with removal of the default environments values.